server/src/main/java/org/apache/cloudstack/backup/BackupManagerImpl.java

```diff
@@ -316,7 +316,8 @@ public boolean restoreVMBackup(final Long backupId, final String restorePointId)
             throw new CloudRuntimeException("VM " + vmId + " couldn't be found on existing or removed VMs");
         }
         accountManager.checkAccess(CallContext.current().getCallingAccount(), null, true, vm);
-        if (vm.getRemoved() == null && !vm.getState().equals(VirtualMachine.State.Stopped)) {
+        if (vm.getRemoved() == null && (!vm.getState().equals(VirtualMachine.State.Stopped) ||
+                !vm.getState().equals(VirtualMachine.State.Destroyed))) {
```

Comment:
@rhtyd this condition is causing an exception for existing VMs in Stopped state. Do destroyed VMs (but not expunged) have their 'removed' field not null at that point? If that's true we can simply replace that 'or' operator for an 'and'. What do you think?